File: HandZ6Api/src/main/java/org/example/handz6api/model/Jugador.java
package org.example.handz6api.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.example.handz6api.model.enums.PosicionJugador;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "jugador")
public class Jugador {
    @Id
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "nombre", nullable = false, length = 100)
    private String nombre;

    @Column(name = "dorsal", nullable = false)
    private Integer dorsal;

    @ManyToOne(fetch = FetchType.LAZY)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "equipo_id")
    private Equipo equipo;

    // Añade en model/Jugador.java
    @Enumerated(EnumType.STRING)
    @Column(name = "posicion")
    private PosicionJugador posicion;


}

File: HandZ6/app/src/main/java/com/example/handz6/data/Entities/Jugador.kt
package com.example.handz6.data.Entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey
import com.example.handz6.data.local.enums.PosicionJugador

// Jugador.kt
@Entity(
    tableName = "jugador",
    foreignKeys = [
        ForeignKey(
            entity = Equipo::class,
            parentColumns = ["id"],
            childColumns = ["equipo_id"],
            onDelete = ForeignKey.SET_NULL
        )
    ],
    indices = [
        Index(value = ["equipo_id"])
    ]
)
data class Jugador(
    @PrimaryKey
    @ColumnInfo(name = "id")
    val id: Long,

    @ColumnInfo(name = "nombre")
    val nombre: String,

    @ColumnInfo(name = "dorsal")
    val dorsal: Int,

    @ColumnInfo(name = "equipo_id")
    val equipoId: Long? = null,

    @ColumnInfo(name = "posicion")
    val posicion: PosicionJugador? = null,
)
